File: Espd/typings/CodeListContract.ts
import { CodeContract } from './CodeContract'
import assign from 'lodash.assign'

interface RequiredArgsCodeListContract {
}

export class CodeListContract {
  public constructor(init?:Partial<CodeListContract> & RequiredArgsCodeListContract) {
    assign(this, init)
  }
  public shortName?: string
  public longName?: string
  public codes?: CodeContract[]
  public agencyId?: string
  public version?: string
  
}

File: Espd/CodeLists/CodeListContract.cs
﻿using System.Collections.Generic;
using Hilma.UBL.Attributes;

namespace Hilma.Espd.EDM.CodeLists
{
    [Contract]
    public class CodeListContract
    {
        public string ShortName { get; set; }
        public string LongName { get; set; }
        public CodeContract[] Codes { get; set; }
        public string AgencyId { get; set; }
        public string Version { get; set; }
    }
}
